libgstc/c/libgstc.c

```diff
@@ -1037,59 +1038,42 @@ gstc_pipeline_signal_connect (GstClient * client, const char *pipeline_name,
   gstc_assert_and_ret_val (NULL != signal, GSTC_NULL_ARGUMENT);
   gstc_assert_and_ret_val (NULL != response, GSTC_NULL_ARGUMENT);
 
+  /* Update the timeout */
   asprintf_ret =
-      asprintf (&what, PIPELINE_SIGNAL_CONNECT_FORMAT, pipeline_name, element,
+      asprintf (&what, PIPELINE_SIGNAL_TIMEOUT_FORMAT, pipeline_name, element,
       signal);
   if (PRINTF_ERROR == asprintf_ret) {
     return GSTC_OOM;
   }
 
-  ret = gstc_cmd_read (client, what, response, client->timeout);
-
-  free (what);
-
-  return ret;
-}
-
-
-GstcStatus
-gstc_pipeline_signal_timeout (GstClient * client, const char *pipeline_name,
-    const char *element, const char *signal, const int value)
-{
-  GstcStatus ret;
-  char *what;
-  char *how;
-  int asprintf_ret;
+  asprintf_ret = asprintf (&how, "%d", timeout);
+  if (PRINTF_ERROR == asprintf_ret) {
+    ret = GSTC_OOM;
+    goto out;
```

Comment:
Out will attempt to free `how`, which is invalid if above fails. you need to add a new label to have the clean single-input/single-output.